lib/Dialect/HW/HWTypes.cpp

```diff
@@ -668,7 +668,9 @@ UnpackedArrayType::verify(function_ref<InFlightDiagnostic()> emitError,
 }
 
 size_t UnpackedArrayType::getNumElements() const {
-  return getSizeAttr().cast<IntegerAttr>().getInt();
+  if (auto intAttr = getSizeAttr().dyn_cast<IntegerAttr>())
+    return intAttr.getInt();
+  return -1;
```

Comment:
Even though Array::getNumElements() is same but probably it's nicer to change `size_t` to `int64_t` if we want to use `-1` as null?

Reply:
If we really want to represent null, we should use std::optional. The trick here is that at least one of the checks (in isZeroBits) is that there's a greater than 0 check so this "just works". I briefly looked into it, but it'd be a more invasive change. So let's just emulate the ArrayType behavior for now.